Wordle.Api/Services/GameService.cs

```diff
@@ -4,142 +4,200 @@
+                                     x.WordDate == date.Value);
+            if (existingGame is not null)
             {
-                // If this is a new game, get a random word.
-                word = GetWord();
+                return existingGame;
             }
+            // If this is a new game, get the word of the day.
+            word = GetDailyWord(date.Value) ?? throw new ArgumentException("Date is too far in the future");
+        }
+        else
+        {
+            // If this is a new game, get a random word.
+            word = GetWord();
+        }
 
-            var game = new Game()
-            {
-                WordId = word.WordId,
-                PlayerId = player.PlayerId,
-                DateStarted = DateTime.UtcNow,
-                GameType = gameType,
-                WordDate = date
-            };
-            _context.Games.Add(game);
+        var game = new Game()
+        {
+            WordId = word.WordId,
+            PlayerId = player.PlayerId,
+            DateStarted = DateTime.UtcNow,
+            GameType = gameType,
+            WordDate = date
+        };
+        _context.Games.Add(game);
 
-            _context.SaveChanges();
+        _context.SaveChanges();
 
-            return game;
+        game.Word = word;
+        return game;
 
-        }
+    }
 
-        public void FinishGame(int gameId)
-        {
-            var game = _context.Games
-                .FirstOrDefault(x => x.GameId == gameId);
-            if (game is null) throw new ArgumentException("Game does not exist");
+    public void FinishGame(int gameId)
+    {
+        var game = _context.Games
+            .FirstOrDefault(x => x.GameId == gameId);
+        if (game is null) throw new ArgumentException("Game does not exist");
 
-            game.DateEnded = DateTime.UtcNow;
-            _context.SaveChanges();
-        }
+        game.DateEnded = DateTime.UtcNow;
+        _context.SaveChanges();
+    }
 
-        public Word GetWord()
+    public Word GetWord()
+    {
+        int wordCount = _context.Words.Count(f => f.Common);
+        int randomIndex = new Random().Next(0, wordCount);
+        Word chosenWord = _context.Words
+            .Where(f => f.Common)
+            .OrderBy(w => w.WordId)
+            .Skip(randomIndex)
+            .Take(1)
+            .First();
+        return chosenWord;
+    }
+
+    public IEnumerable<DateWord?> GetLast10DateWords()
+    {
+        for (var i = 0; i < 10; i++)
         {
-            int wordCount = _context.Words.Count(f => f.Common);
-            int randomIndex = new Random().Next(0, wordCount);
-            Word chosenWord = _context.Words
-                .Where(f => f.Common)
-                .OrderBy(w => w.WordId)
-                .Skip(randomIndex)
-                .Take(1)
-                .First();
-            return chosenWord;
+            yield return GetDailyDateWord(DateTime.UtcNow.AddDays(-i));
```

Comment:
This isn't really an appropriate use of a yield return, and in general, there aren't really any great spots in the app as it stands for a yield return. What should be happening here instead is a query to the database should be happening that uses the `Take(10)` method.